torch/csrc/jit/codegen/cuda/ir_base_nodes.cpp

```diff
@@ -262,6 +204,25 @@ Expr::Expr(ExprType _type) : type_{_type} {
   this->fusion_ = fusion;
 }
 
+Expr::Expr(const Expr* src, IrCloner* ir_cloner)
+    : Statement(src, ir_cloner),
+      type_(src->type_),
+      inputs_(ir_cloner->clone(src->inputs_)),
+      outputs_(ir_cloner->clone(src->outputs_)) {}
+
+bool Expr::sameAs(const Expr* const other) const {
```

Comment:
I realize this is my fault, but I don't think this is safe to call. Realizing binaryOp, unaryOp, and ternaryOp won't evaluate correctly here. Would you mind just returning `this==other`? as I think that's the best we can do without implementing dispatch based checking here.

Reply:
I'm not sure I follow. There are no new calls to sameAs() in this PR, it's been just moved out of the header (not only to simplify the class definition, but it also allows calling the auto-return functions like Expr::inputs() w/o any concerns regarding the declaration ordering)

Reply:
This function is bad and my fault. Was just wondering since you touched it if you would simplify it to something that would actually be more correct i.e. `return this==other;`

Reply:
Oh, I see. Having a function which just does `this == other` doesn't seem to be worth it, so we may want to consider just replacing sameAs() directly with == where it makes sense.

I opened https://github.com/csarofeen/pytorch/issues/131 to track it.